### PR TITLE
BOM-2651: Django 3.2 Support

### PR DESCRIPTION
**Pull request by mzulqarnain1**
## Description

- Added Django 3.2 Support
- Applied Django Code Mods & Pyupgrade 
- Ran Make Upgrade

## Ticket Link

Link to the associated ticket
https://openedx.atlassian.net/browse/BOM-2651

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-20.04
+    strategy:
+      max-parallel: 4
```
Comment (mraarif) [inline .github/workflows/ci.yml]: do we need to restrict it to maximum of 4 parallel runners?

File: xqueue/docker_gunicorn_configuration.py [hunk 2]
```diff
@@ -7,7 +7,7 @@
 workers = 6
 
 def pre_request(worker, req):
-    worker.log.info("%s %s" % (req.method, req.path))
+    worker.log.info(f"{req.method} {req.path}")
```
Comment (mraarif) [inline xqueue/docker_gunicorn_configuration.py]: we shouldn't be using `f` strings with log statements, we can instead use `format`

**Comment by mraarif** [comment]
we should remove Travis support in a subsequent PR

File: xqueue/docker_gunicorn_configuration.py (hunk 2 above)
Comment (mzulqarnain1) [reply xqueue/docker_gunicorn_configuration.py]: Changed to format.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (mzulqarnain1) [reply .github/workflows/ci.yml]: Removed this restriction.

**Review by mraarif — APPROVED** [review]

**Comment by natabene** [comment]
@awais786 @mzulqarnain1 Owning team doesn't need to review. They selected Hands Off option, which means they leave all the work to us and are not planning to review.

**Review by UsamaSadiq — APPROVED** [review]

**Pull request merged**